src/rpc/rawtransaction.cpp

```diff
@@ -2376,6 +2406,36 @@ static RPCHelpMan analyzepsbt()
     }
     if (!inputs_result.empty()) result.pushKV("inputs", inputs_result);
 
+    UniValue outputs_result(UniValue::VARR);
+    for (const auto& output : psbtx.outputs) {
+        UniValue output_univ(UniValue::VOBJ);
```

Comment:
Good idea!

Reply:
Actually I think the existing "next role" logic is fine. Having invalid/missing proofs does not affect whether the transaction should be considered fully blinded or not.

But I will move the analysis to node/psbt.cpp anyway because it makes more conceptual sense there.